identity/src/commonMain/kotlin/com/android/identity/nfc/NdefRecord.kt

```diff
@@ -87,19 +91,21 @@ data class NdefRecord(
         if (isShortRecord) flags = flags or FLAG_SR
         if (idLengthPresent) flags = flags or FLAG_IL
 
-        buf.writeByte(flags.toByte())
-        buf.writeByte(type.size.toByte())
+        bsb.appendUInt8(flags)
+        bsb.appendUInt8(type.size)
         if (isShortRecord) {
-            buf.writeByte(payload.size.toByte())
+            bsb.appendUInt8(payload.size)
         } else {
-            buf.writeInt(payload.size)
+            bsb.appendUInt32(payload.size) // TODO: b/393388370 - is it really 4 byte? or perhaps it's 2 (UInt16)?
         }
         if (idLengthPresent) {
-            buf.writeByte(id.size.toByte())
+            bsb.appendUInt8(id.size)
         }
-        buf.write(type.toByteArray())
-        buf.write(id.toByteArray())
-        buf.write(payload.toByteArray())
+        // TODO: b/393388370 - it looks like a common issue with different ByteArray behaviour for empty data.
```

Comment:
Is the error that ByteString.toByteArray() throws an exception if it's empty? That's a little unclear with this comment (and the bug doesn't provide any background on this particular issue).

Is there a reason we need to convert it to a ByteArray? It looks like there's an existing ByteStringBuilder method to append() a ByteString.

And I'm wondering if this is a temporary error that will go away when we migrate more of the code toward ByteStrings...?